frame/system/src/weights.rs

```diff
@@ -16,20 +16,18 @@
 // limitations under the License.
 
 use codec::{Encode, Decode};
-use frame_support::weights::{Weight, DispatchClass};
-use sp_runtime::RuntimeDebug;
+use frame_support::weights::{Weight, DispatchClass, constants};
+use sp_runtime::{RuntimeDebug, Perbill};
 
 /// An object to track the currently used extrinsic weight in a block.
-#[derive(Clone, Eq, PartialEq, Default, RuntimeDebug, Encode, Decode)]
-pub struct ExtrinsicsWeight {
-	normal: Weight,
-	operational: Weight,
-}
+pub type ExtrinsicsWeight = ForDispatchClass<Weight>;
```

Comment:
any reason why we have this type alias? `impl` on an alias seems a bit strange.

Reply:
I would name `ForDispatchClass` something that clearly says this is accumulating the weight of a _block_.

Reply:
Okay I see that it is actually used to note the weight of a single extrinsic as well, like in the fields of `BlockWeights`. Then the name is sensible, perhaps I would just name it `WeightPerClass<T>` since it is a struct that always holds one value per class.